src/comp/ShareButton.tsx

```diff
@@ -59,9 +58,9 @@ const ShareButton = (props: ShareButtonProps) => {
     <>
       <div className={styles.buttonGroup}>
         <CopyUrl url={pageData.url}>
-          <FontAwesomeIcon icon={iconsUrl["Copy"].definition} />
+          <FontAwesomeIcon icon={iconsUrl.Copy.definition} />
         </CopyUrl>
-        {iconList.map((icon: keyof Icon, key: number) => (
+        {iconList.map((icon: IconName, key: number) => (
```

Comment:
```suggestion
        {icons.map((icon: Icon, key: number) => (
```
`ShareButton` も `IconName` のリストではなく `Icon` のリストを扱いたい
`mainIconName: IconName[]` と `subIconName: IconName[]` を使うんじゃなくて、
`mainIcon: Icon[]` と `subIcon: Icon[]` を `data.tsx` に提供してもらってそれを使う方がいいと思う